docs/docs/using-vite.md

```diff
@@ -0,0 +1,140 @@
+
+export default ({ site }: { site: string }) => {
+  hydrateRoot(
++    document.getElementById('root')!,
+-    document.getElementById('app')!,
+    <BrowserRouter>
+      <App site={site} />
+    </BrowserRouter>,
+  );
+};
+```
+
+### Render entrypoint
+
+The render entrypoint file has to export a `ViteRender` object. The `ViteRender` object has two functions `render` and `provideClientContext`.
+
+**render**
+
+The `render` function should return your application as a pipeable stream (using `React.renderToPipeableString`).
+
+`render` will be called once for each combination of settings in sku config. Specifically, `environment`, `site` & `route`.
+
+If you are using `loadable` components, you should wrap your application in a `LoadableProvider` from `sku/vite/loadable`. This will search your rendered application for dynamic imports that are wrapped in a `loadable` function.
+By calling the `await preloadAll()` function, you can ensure that all `loadable` imported files will be preloaded before sku renders the page.
+
+**provideClientContext**
+
+`provideClientContext` is an optional function that runs after `render`. It is used to pass context between the static render and the client code. This is useful for passing config values (e.g. API endpoints, feature switches) or state information (e.g. redux state). The object this function returns will be passed to the client entry.
+
+The function receives `environment` and `site`
+
+**Example of a render entrypoint with `vite`**
+
+```tsx
+import { StaticRouter } from 'react-router-dom/server';
+import type { ViteRender } from 'sku';
+import { renderToPipeableStream } from 'react-dom/server';
+import { LoadableProvider, preloadAll } from 'sku/vite/loadable';
+
+import App from './App';
+
+export default {
+  render: async ({ options, renderContext, site, url }) => {
+    const { loadableCollector } = renderContext;
+
+    await preloadAll();
+
+    return renderToPipeableStream(
+      <LoadableProvider value={loadableCollector}>
+        <StaticRouter location={url} context={{}}>
+          <App site={site} />
+        </StaticRouter>
+      </LoadableProvider>,
+      options,
+    );
+  },
+
+  provideClientContext: ({ site }) => ({
+    site,
+  }),
+} satisfies ViteRender;
+```
+
+### Code splitting
+
+Code splitting is possible with the new `sku/vite/loadable` entrypoint.
+
+The new `sku/vite/loadable` entrypoint relies on React's [`<Suspense />`](https://react.dev/reference/react/Suspense) component for the loading of a fallback state.
+
+**Example of using `sku/vite/loadable`**
+
+```tsx
+import { Suspense } from 'react';
+import { loadable } from 'sku/vite/loadable';
+
+const Home = loadable(() => import('./Home'));
+
+export default () => (
+  <div>
+    <Suspense fallback={<div>Loading Home...</div>}>
+      <Home />
+    </Suspense>
+  </div>
+);
+```
+
+### Using vite types
+
+If `vite/client` reference types are needed in your project, you can import them from `sku/vite/client`.
```

Comment:
```suggestion
If [`vite/client` types](https://vite.dev/guide/features#client-types) are needed in your project, you can reference them via the `sku/vite/client` entrypoint:
```

Should we also be more specific about where to put this `<reference />`? Maybe `client.tsx`?